### PR TITLE
pokemon-component-stuff

### PR DESCRIPTION
**Pull request by j86hughes**

### CODE REVIEW

**Review by Chris-Kasatka — APPROVED** [review]

**Review by Chris-Kasatka — APPROVED** [review]

**Comment by j86hughes** [comment]
mate delete `package-lock.json`

`package-lock.json` gets made by using `npm` so probably cos you did `npm install`.

Problem is mate we're using `yarn` rather than `npm` for this project. `npm` and `yarn` are package managers that do the same thing but you should stick to only using one per project. `package-lock.json` basically keeps a record of all the packages your project is using and their versions, but `yarn.lock` is yarn's version which does the same thing.

So in future mate do `yarn install` rather than `npm install` (for this project)

File: src/pages/Pokemon/Nav.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { useLocation } from "react-router-dom";
+import './pokemon.css'
+
+const Nav = () => {
+    const { state } = useLocation();
```
Comment (j86hughes) [inline src/pages/Pokemon/Nav.js:6]: I know this component it just a dummy but let's remove this `useLocation` hook anyways to make it cleaner 👍

File: src/pages/Pokemon/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { useLocation } from "react-router-dom";
+import './pokemon.css'
+
+const Pagination  = () => {
+    const { state } = useLocation();
```
Comment (j86hughes) [inline src/pages/Pokemon/Pagination.js:6]: same as comment above but yeah let's remove this hook

File: src/pages/Pokemon/Stats.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import { useLocation } from "react-router-dom";
+import './pokemon.css'
+
+
+const Stats = () => {
+  const { state } = useLocation();
+  const pokemonItem = state?.pokemonItem;
```
Comment (j86hughes) [inline src/pages/Pokemon/Stats.js:8]: mate you can PURGE this and then get `pokemonItem` from the props

File: src/pages/Pokemon/Stats.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+      </div>
+      </div>
+    )
+}
+
+export default Stats
+
+
+
+{/* <ul>
```
Comment (j86hughes) [inline src/pages/Pokemon/Stats.js:33]: maybe delete all this commented code?

File: src/pages/Pokemon/Statssmall.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                item.type.name.slice(1)}
+            </div>
+          );
+        })}
+      </div>
+      </div>
+    )
+}
+
+export default Statssmall
```
Comment (j86hughes) [inline src/pages/Pokemon/Statssmall.js:29]: need extra line space

File: src/pages/Pokemon/index.js [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
+import Nav from "./Nav";
+import Pagination from "./Pagination";
+import Stats from "./Stats";
+import Image from "./Image";
+import Info from "./Info";
 
 const Test = () => {
-  let params = useParams();
+  const { state } = useLocation();
+  const pokemonItem = state?.pokemonItem;
```
Comment (j86hughes) [inline src/pages/Pokemon/index.js:14]: mate you can PURGE this and then YAALALAALALALALALALALALALALALALALALALALALALALALALALA

File: src/pages/Pokemon/index.js [hunk 7]
```diff
@@ -1,11 +1,31 @@
...
+  console.log(pokemonItem);
```
Comment (j86hughes) [inline src/pages/Pokemon/index.js:15]: we can get rid of this console.log

File: src/pages/Pokemon/Nav.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './pokemon.css'
+
+const Nav = ({ pokemonItem }) => {
```
Comment (j86hughes) [inline src/pages/Pokemon/Nav.js:4]: I know this is a dummy component but we probs don't wanna pass `pokemonItem` here though mate

File: src/pages/Pokedex/Pokedex.js [hunk 9]
```diff
@@ -11,6 +11,71 @@ const getPokemonDetails = async (pokemon) => {
...
+  P.getGenderByName("female")
+.then((gender1) => {
+  // console.log(gender1.name);
+  console.log(gender1);
+})
+.catch((error) => {
+  console.log('There was an ERROR: ', error);
+});
+}
+getGender()
```
Comment (j86hughes) [inline src/pages/Pokedex/Pokedex.js:51]: mate yeah probs don't wanna run this every time, so let's comment this line out

File: src/pages/Pokemon/Statssmall.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import './pokemon.css'
+
+
+const Statssmall = (pokemonItem) => {
```
Comment (j86hughes) [inline src/pages/Pokemon/Statssmall.js:5]: mate sorry, can you rename this component and the file to `StatsSmall` - it's best practice to name things in camel case mate

**Pull request merged**